emscripten/native/ui/EmscriptenInput.cpp

```diff
@@ -0,0 +1,228 @@
+#include "EmscriptenInput.h"
+#include "ui/MapView.h"
+#include <unordered_map>
+
+namespace carto {
+    static const int NATIVE_ACTION_POINTER_1_DOWN = 0;
+    static const int NATIVE_ACTION_POINTER_2_DOWN = 1;
+    static const int NATIVE_ACTION_MOVE = 2;
+    static const int NATIVE_ACTION_CANCEL = 3;
+    static const int NATIVE_ACTION_POINTER_1_UP = 4;
+    static const int NATIVE_ACTION_POINTER_2_UP = 5;
+    static const int NATIVE_NO_COORDINATE = -1;
+
+    struct Coord {
+        long x;
+        long y;
+    };
+
+    std::unordered_map<long, Coord> touchList;
```

Comment:
Perhaps this code could be moved into a separate class (`EmscriptenInput`). Current implementation populates `carto` namespace with lots of additional variables.

Reply:
I added additional `EmscriptenInput` namespace. Is this okay or would it be better to create a class?